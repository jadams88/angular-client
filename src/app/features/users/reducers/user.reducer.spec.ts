import { usersReducer, UsersEntityState } from '~/app/features/users/reducers/user.reducer';
import { User } from '~/app/features/users/models/user.model';
import {
  LoadUsersSuccess,
  UpdateUserSuccess,
  DeleteUserSuccess
} from '~/app/features/users/actions/user.actions';

describe('UsersReducer', () => {
  describe('undefined action', () => {
    it('should return the default state', () => {
      const action = {} as any;
      const result = usersReducer(undefined, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('LoadSucces action', () => {
    it('should add the user to the user state', () => {
      const users = [
        { id: '2', username: 'someUser', emailAddress: 'some@emailAddress.com' }
      ] as User[];
      const action = new LoadUsersSuccess(users);

      const result = usersReducer(undefined, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('UpdateSucces action', () => {
    it('should update the existing user state', () => {
      const user = {
        id: '1',
        username: 'initialUser',
        emailAddress: 'some@emailAddress.com'
      } as User;

      const initialState: UsersEntityState = {
        ids: [user.id],
        entities: {
          [user.id]: user
        },
        selectedUserId: null,
        authenticatedUserId: null
      };

      const userToUpdate = { id: '1', username: 'updatedName' } as User;
      const action = new UpdateUserSuccess(userToUpdate);

      const result = usersReducer(initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('DeleteSucces action', () => {
    it('should delete the existing user state', () => {
      const user = {
        id: '1',
        username: 'initialUser',
        emailAddress: 'some@emailAddress.com'
      } as User;

      const initialState: UsersEntityState = {
        ids: [user.id],
        entities: {
          [user.id]: user
        },
        selectedUserId: null,
        authenticatedUserId: null
      };

      const userToDelete = { id: '1' } as User;

      const action = new DeleteUserSuccess(userToDelete);

      const result = usersReducer(initialState, action);

      expect(result).toMatchSnapshot();
    });
  });
});
