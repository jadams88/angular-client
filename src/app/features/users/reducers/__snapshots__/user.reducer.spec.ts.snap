// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersReducer DeleteSucces action should delete the existing user state 1`] = `
Object {
  "authenticatedUserId": null,
  "entities": Object {},
  "ids": Array [],
  "selectedUserId": null,
}
`;

exports[`UsersReducer LoadSucces action should add the user to the user state 1`] = `
Object {
  "authenticatedUserId": null,
  "entities": Object {
    "2": Object {
      "emailAddress": "some@emailAddress.com",
      "id": "2",
      "username": "someUser",
    },
  },
  "ids": Array [
    "2",
  ],
  "selectedUserId": null,
}
`;

exports[`UsersReducer UpdateSucces action should update the existing user state 1`] = `
Object {
  "authenticatedUserId": null,
  "entities": Object {
    "1": Object {
      "emailAddress": "some@emailAddress.com",
      "id": "1",
      "username": "updatedName",
    },
  },
  "ids": Array [
    "1",
  ],
  "selectedUserId": null,
}
`;

exports[`UsersReducer undefined action should return the default state 1`] = `
Object {
  "authenticatedUserId": null,
  "entities": Object {},
  "ids": Array [],
  "selectedUserId": null,
}
`;
